pkg/controllers/dashboard/systemcharts/controller.go

```diff
@@ -33,20 +35,30 @@ func Register(ctx context.Context, wContext *wrangler.Context, registryOverride
 	}
 
 	wContext.Catalog.ClusterRepo().OnChange(ctx, "bootstrap-charts", h.onRepo)
-	relatedresource.WatchClusterScoped(ctx, "bootstrap-charts", func(namespace, name string, obj runtime.Object) ([]relatedresource.Key, error) {
+	relatedresource.WatchClusterScoped(ctx, "bootstrap-charts", func(_, _ string, _ runtime.Object) ([]relatedresource.Key, error) {
 		return []relatedresource.Key{{
 			Name: repoName,
 		}}, nil
 	}, wContext.Catalog.ClusterRepo(), wContext.Mgmt.Feature())
 
-	relatedresource.WatchClusterScoped(ctx, "bootstrap-settings-charts", func(namespace, name string, obj runtime.Object) ([]relatedresource.Key, error) {
+	relatedresource.WatchClusterScoped(ctx, "bootstrap-settings-charts", func(_, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
 		if s, ok := obj.(*v3.Setting); ok && (s.Name == "rancher-webhook-version" || s.Name == "rancher-webhook-min-version") {
 			return []relatedresource.Key{{
 				Name: repoName,
 			}}, nil
 		}
 		return nil, nil
 	}, wContext.Catalog.ClusterRepo(), wContext.Mgmt.Setting())
+
+	// ensure the system charts are installed with the correct values when there are changes to the rancher config map
+	relatedresource.WatchClusterScoped(ctx, "bootstrap-configmap-charts", func(_, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
```

Comment:
I'm concerned about the performance of this function. This check will essentially be run anytime that a configmap is altered in a cluster. For downstream clusters this could happen thousands of times in a very short period of time.

Is there any way that we can prevent this watch from being evaluated unless the configmap is in cattle-system?